src/main/java/picard/sam/SamErrorMetric/CollectSamErrorMetrics.java

```diff
@@ -309,6 +370,33 @@ protected int doWork() {
         return 0;
     }
 
+    /**
+     * Creates one file for each filter, each line representing the ID of a matched read and the number of occurrences
+     * that this read had matched the filter. Subsequently, all reads that meet any filter are written to a BAM file
+     * which can be queried by the read ID found in the filter output files.
+     * @param samFileHeader Header to use for the output BAM file
+     */
+    private void writeOutputReads(SAMFileHeader samFileHeader) {
+        try {
+            SAMFileWriter writer = new SAMFileWriterFactory()
+                    .setCompressionLevel(2)
+                    .makeBAMWriter(samFileHeader, false, new File(OUTPUT + ".outputreads.bam"));
+            outputReadsById.values().forEach(writer::addAlignment);
+            writer.close();
+
+            for(SamErrorReadFilter filter : samErrorReadFilters) {
```

Comment:
what is going on here? what are you writing? where is this behavior described? very confused.